File: backend/src/main/java/de/bund/digitalservice/ris/caselaw/domain/court/Court.java
package de.bund.digitalservice.ris.caselaw.domain.court;

import de.bund.digitalservice.ris.caselaw.domain.DocumentationOffice;
import java.util.List;
import java.util.UUID;
import lombok.Builder;

@Builder
public record Court(
    UUID id,
    String type,
    String location,
    String label,
    String revoked,
    String jurisdictionType,
    List<String> regions,
    DocumentationOffice responsibleDocOffice,
    List<CourtBranchLocation> courtBranchLocations,
    Boolean isSuperiorCourt,
    Boolean isForeignCourt) {}

File: frontend/src/domain/court.ts
import { CourtBranchLocation } from "@/domain/courtBranchLocation"
import DocumentationOffice from "@/domain/documentationOffice"

export type Court = {
  type?: string
  location?: string
  label: string
  revoked?: string
  jurisdictionType?: JurisdictionType
  regions?: string[]
  responsibleDocOffice?: DocumentationOffice
  isSuperiorCourt?: boolean
  isForeignCourt?: boolean
  courtBranchLocations?: CourtBranchLocation[]
}

export type JurisdictionType =
  | "Arbeitsgerichtsbarkeit"
  | "Besondere Gerichtsbarkeit"
  | "Finanzgerichtsbarkeit"
  | "Ordentliche Gerichtsbarkeit"
  | "Sozialgerichtsbarkeit"
  | "Verfassungsgerichtsbarkeit"
  | "Verwaltungsgerichtsbarkeit"
